lightningd/peer_htlcs.c

```diff
@@ -590,16 +592,194 @@ static void channel_resolve_reply(struct subd *gossip, const u8 *msg,
 	tal_free(gr);
 }
 
+/**
```

Comment:
Nit : Perhaps all the hook structures / functions can live in another file? It seems there are many options for future hooks and I can imagine them bloating some files that are already quite large, so it might set the precedent to keep those in their own space.

Reply:
I considered something similar, but most of the logic is actually either taken from the `peer_htlcs.c` file (in the callback) or wraps and unwraps data that may not be available elsewhere, which is why I ended up putting it here. I'd prefer to keep the logic at least in the same place.